lib/src/exception_handler/exceptions.dart

```diff
@@ -0,0 +1,245 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+part of angular_ast.src.exceptions;
+
+/// Error codes used for exceptions that occur during the parsing.
+/// The convention for this class is for the name of the error code
+/// to indicate the problem that caused the error code to be generated
+/// and for the error message to explain what is wrong, and when appropriate,
+/// how the problem can be corrected.
+class NgParserWarningCode extends ErrorCode {
+  static const NgParserWarningCode AFTER_COMMENT = const NgParserWarningCode(
+    'AFTER_COMMENT',
```

Comment:
Yeah, I initially set the name of the errors as the state of where they're thrown. They'll have to be refactored into more meaningful names. I also want to relook at the exception overlaps between angular_ast and angular_analyzer_plugin in a future rework of exception codes.